src/main/scala/com/twitter/diffy/proxy/SequentialMulticastService.scala

```diff
@@ -3,15 +3,95 @@ package com.twitter.diffy.proxy
 import com.twitter.finagle.Service
 import com.twitter.util.{Future, Try}
 
-class SequentialMulticastService[-A, +B](
-    services: Seq[Service[A, B]])
+import org.jboss.netty.handler.codec.http.HttpRequest
```

Comment:
This generic provides a common protocol-agnostic abstraction to multicast something to an arbitrary number of destinations. The concern of having target specific headers belongs to the HttpDifferenceProxy implementation and should live there.

Further, instead of creating 3 different versions of the request before sending any of them to the targets perhaps you could compose each target with a Filter that independently re-writes whatever requests it receives before forwarding it to the underlying target.